substrate-archive/src/actors/workers/storage_aggregator.rs

```diff
@@ -51,6 +53,19 @@ where
 		}
 		Ok(())
 	}
+
+	async fn handle_traces(&mut self, ctx: &mut Context<Self>) -> Result<()> {
+		let mut traces = std::mem::take(&mut self.traces);
+		if !traces.is_empty() {
+			log::info!("Inserting {} traces", traces.len());
+			for trace in traces.drain(..) {
+				let send_result = self.db.send(trace.into()).await?;
+				ctx.handle_while(self, send_result).await;
+			}
+		}
+		std::mem::swap(&mut self.traces, &mut traces);
```

Comment:
I don't understand why `swap` is performed here,  `traces` should be empty anyways after `drain(..)`

Reply:
This is form my review. It keeps the capacity of the vector (`drain` removes the items, but it doesn't deallocate) without having to reallocate a new `Vec` when you take ownership of it.

Reply:
ok, I see.